### ISSUE TITLE
Relaxing the constraint on the dimensions of the output and true label

### ISSUE THREAD

**Issue by M-Taha** [fchollet/keras#1048]
Apparently, Keras requires the shapes of the true output `y_true` and the predicted output `y_pred` to  match.  Is it possible to relax this constraint?  There are many loss functions that require different shapes for these two.

**Comment by fchollet** [comment]
If you write your own loss functions, you can do whatever. I don't get what you're proposing?

**Comment by EderSantana** [comment]
note that although both have to match, you can always pad the desired with zeros and in your own cost function just use the part of it you want.

**Comment by fchollet** [comment]
Your statement was initially confusing me because it is incorrect (the shapes don't have to match, actually I've often written loss functions where the two had completely different shapes). I think what you meant is that the first dimension (the samples dimension) of each tensor has to match. Is that right?

If you think about it is should be apparent why that is the case.

**Comment by M-Taha** [comment]
@fchollet Thank you for the answer and I am absolutely sorry for taking your time.  
In order to demonstrate the issue more clearly, I wrote a bare minimum code and realized that it works.  Perhaps, there is a bug in my other code.
You may close this thread, but in order to keep it useful for future readers, I put my example here.  

Please feel free to add my objective function to the list of objective functions because it is an important one in statistics.
____
## Survival Analysis with Keras
Let's design a simple model for survival analysis in Keras.  I will use the exponential hazard rate and the likelihood in Eq. (7.7) in [this note](http://data.princeton.edu/wws509/notes/c7.pdf).  Write the following loss function and add it to `/keras/kera/objectives.py`:
```python
def survival(y_true, y_pred):
    # y_true[:,0]: times, y_true[:,1]: deaths, y_pred[:,0]: log of hazard rate
    return T.mean(T.exp(y_pred[:,0])*y_true[:,0] - y_true[:,1]*y_pred[:,0], axis=-1)
```
Using this loss function, we can write the main script as follows:
```python
from keras.models import Sequential  
from keras.layers.core import Dense
import numpy as np

inputDim = 10
nSample = 1000

def generateData():  
    np.random.seed(0)  # For reproducability
    features = np.random.standard_normal((nSample,inputDim))
    death = np.random.randint(0,1,(nSample,1))
    time = np.random.exponential(1,(nSample,1))
    return features, np.hstack((time,death))

def buildModel():
    # A linear regression model to predict logarithm of the hazard rate
    # You can make it a lot more complex
    model = Sequential()  
    model.add(Dense(1, input_dim=inputDim, activation='linear'))
    model.compile(loss="survival", optimizer="rmsprop") 
    return model

if __name__ == "__main__":
    model = buildModel()
    X,y = generateData()
    model.fit(X, y, validation_split=0.1)  
```
